docs/encodings/anndata_encoding.md

```diff
@@ -0,0 +1,90 @@
+# cellxgene Data Integration Schema AnnData Implementation
+​
+Authors: acarr@chanzuckerberg.com
+​
+Document Status: _Approved_
+​
+Version: 0.2.0
+​
+Date Last Modified: 2020-11-02
+​
+cellxgene Data Portal datasets are stored in the HDF5-backed AnnData format as written by version 0.7 of the anndata
+library. One goal of using this format is to have all metadata stored in the same file as the data, so cellxgene will
+have access to all metadata within a single file.
+​
+The schema requirements and definitions for the AnnData `X`, `uns`, `obs`, and `obsm` attributes are below.
+​
+An h5ad that follows the schema will have a `version` attribute in `uns` which is a `dict`, and that dict will
+have a key "corpora_schema_version".
+​
+### Note on types
+The types below are python3 types. Note that a python3 `str` is a sequence of Unicode code points, which is stored
+null-terminated and UTF-8-encoded by anndata.
+​
+​
+## `X`
+​
+cellxgene does not impose any additional constraints on the `X` data matrix. So it may be sparse or dense and any
+numeric `numpy.dtype`.
+​
+The cellxgene schema has different requirements for different assay types.
+
+| Assay                                 | "raw" required? | "raw" location                                                      | "final" location | "final" required? | Other layers |
+|---------------------------------------|-----------------|---------------------------------------------------------------------|------------------|-----------------------|--------------|
+| scRNA-seq (UMI, e.g. 10x v3)          | REQUIRED        | AnnData.layers["raw"] unless no "final" is provided, then AnnData.X | AnnData.X        | STRONGLY RECOMMENDED  | OPTIONAL     |
+| scRNA-seq (non-UMI, e.g. SS2)         | REQUIRED        | AnnData.layers["raw"] unless no "final" is provided, then AnnData.X | AnnData.X        | STRONGLY RECOMMENDED  | OPTIONAL     |
+| Accessibility (e.g. ATAC-seq, mC-seq) | NOT REQUIRED    |                                                                     | AnnData.X        | REQUIRED              | OPTIONAL     |
+| CITE-seq                              | REQUIRED        | AnnData.layers["raw"] unless no "final" is provided, then AnnData.X | AnnData.X        | STRONGLY RECOMMENDED  | OPTIONAL     |
+​
+## `uns`
+​
+Recall that `uns` is a mapping with `str`s as keys. The cellxgene data integration schema requires the following keys and values in `uns`:
+​
+**Key**|**Value Type**|**Notes**
+-----|-----|-----
+version|`dict` with keys "corpora_schema_version" and "corpora_encoding_version" indicating the version of the schema and schema encoding used to create the h5ad.
+title|`str`|
+layer\_descriptions|`dict` with string keys and values|One key must be "X" which describes the transformations (if any) performed to produce the X matrix cellxgene displays.
+organism|`str`|
+organism\_ontology\_term\_id|`str`|
+​
```

Comment:
Need an extra line break here to end the table.

Reply:
Addressed in a17f0d5